databases/week1/Week1-homework.sql

```diff
@@ -0,0 +1,91 @@
+/* Find out how many tasks are in the task table */
+select
+  COUNT(id)
+from
+  task;
+
+/* Find out how many tasks in the task table do not have a valid due date */
+select
+  COUNT(id)
+from
+  task
+where
+  IsNull(due_date);
+
+/* Find all the tasks that are marked as done */
+select
+  *
+from
+  task
+where
+  status_id = 3;
+
+/* Find all the tasks that are not marked as done */
+select
+  *
+from
+  task
+where
+  status_id != 3;
```

Comment:
Here I can't see in which status a task is as I (any external viewer) will not understand (not easily if at least) what status_id=3 meant. We don't even have a status name in the table `task `. So we need to get some information from other related tables (by joining).
Check this if it can help---

```
SELECT task.id,title,due_date, status.name
FROM task
JOIN status ON task.status_id=status.id
WHERE status.name="Done";
```
-- Find all the tasks that are not marked as not done
```
SELECT task.id,title,due_date, status.name
FROM task
JOIN status ON task.status_id=status.id
WHERE status.name!="Done";
```

Reply:
Sure Elias

I will improve my code next time. I will fix it now as the task is small.